lib/spack/spack/version.py

```diff
@@ -1113,11 +1218,11 @@ def lookup_commit(self, commit):
             # won't properly update the local rev-list)
             self.fetcher.git("fetch", '--tags')
 
-            # Ensure commit is an object known to git
-            # Note the brackets are literals, the commit replaces the format string
-            # This will raise a ProcessError if the commit does not exist
+            # Ensure ref is a commit object known to git
+            # Note the brackets are literals, the ref replaces the format string
+            # This will raise a ProcessError if the ref does not exist
             # We may later design a custom error to re-raise
-            self.fetcher.git('cat-file', '-e', '%s^{commit}' % commit)
+            self.fetcher.git('cat-file', '-e', '%s^{commit}' % ref)
```

Comment:
I was playing around with this a little more, and I think it might be a good time to add a custom error. If I knowingly give it a wrong git ref (but a not unreasonable one) we get a rather unclear error message:

```console
$ spack stage yaml-cpp@git.develop
From https://github.com/jbeder/yaml-cpp
 * branch            HEAD       -> FETCH_HEAD
fatal: Not a valid object name develop^{commit}
==> Error: Command exited with status 128:
'/usr/bin/git' '-c' 'advice.detachedHead=false' 'cat-file' '-e' 'develop^{commit}'
```

I don't feel like this is a show stopper though so I won't request holding up the PR over it.  However, I think if not in this PR then soon. 🙂

Reply:
```
$ spack stage yaml-cpp@git.develop
==> Error: develop is not a valid git ref for yaml-cpp
```
better?

Reply:
💯